pulsar-broker/src/main/java/org/apache/pulsar/compaction/PulsarTopicCompactionService.java

```diff
@@ -106,6 +108,45 @@ public CompletableFuture<Position> getLastCompactedPosition() {
         return CompletableFuture.completedFuture(compactedTopic.getCompactionHorizon().orElse(null));
     }
 
+    @Override
+    public CompletableFuture<Position> findNewestPosition(Predicate<Entry> condition) {
+        var compactedTopicContextFuture = compactedTopic.getCompactedTopicContextFuture();
```

Comment:
Is it method thread-safe?

Reply:
CompactedTopicImpl.compactedTopicContext is a volatile variable